src/engine/SCons/Node/FS.py

```diff
@@ -2470,11 +2484,37 @@ def __setstate__(self, state):
             if key not in ('__weakref__',):
                 setattr(self, key, value)
 
+    def __eq__(self, other):
+        return self.csig == other.csig and self.timestamp == other.timestamp and self.size == other.size
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
 
 class FileBuildInfo(SCons.Node.BuildInfoBase):
-    __slots__ = ()
+    """
+    This is info loaded from sconsign.
+
+    Attributes unique to FileBuildInfo:
+        dependency_map : Caches file->csig mapping
+                    for all dependencies.  Currently this is only used when using
+                    MD5-timestamp decider.
+                    It's used to ensure that we copy the correct
+                    csig from previous build to be written to .sconsign when current build
+                    is done. Previously the matching of csig to file was strictly by order
+                    they appeared in bdepends, bsources, or bimplicit, and so a change in order
+                    or count of any of these could yield writing wrong csig, and then false positive
+                    rebuilds
+    """
+    __slots__ = ('dependency_map')
     current_version_id = 2
 
+    def __setattr__(self, key, value):
+        if key != 'dependency_map' and hasattr(self, 'dependency_map'):
```

Comment:
Since this is unusual (delete an attr when any _other_ attr is set), maybe add a comment?